From: David Bremner <bremner@unb.ca>
Subject: libtoolize-lib


---
 lib/Makefile |   21 +++++++++++++++------
 1 files changed, 15 insertions(+), 6 deletions(-)

diff --git a/lib/Makefile b/lib/Makefile
index 7c59549..0ee4adb 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -1,5 +1,11 @@
-#CC = gcc -Wall 
-#RANLIB = echo
+LD=ld
+CC = gcc
+RANLIB = echo
+CFLAGS ?= -Wall -O2 -g
+
+.SUFFIXES: .lo
+.c.lo:
+	 libtool --mode=compile $(CC) -c $(CFLAGS) -o $@ $<
 
 BIB_OBJS = bbl.o fields.o list.o name.o title.o reftypes.o
 
@@ -20,13 +26,16 @@ OUTPUT_OBJS   = bibtexout.o endout.o isiout.o modsout.o risout.o wordout.o \
 BIBUTILS_OBJS = $(SIMPLE_OBJS) $(NEWSTR_OBJS) $(CONTAIN_OBJS) \
 	$(INPUT_OBJS) $(OUTPUT_OBJS) name.o title.o bibl.o serialno.o bibutils.o
 
+BIBUTILS_LO=$(subst .o,.lo,$(BIBUTILS_OBJS))
 BIBPROGS_OBJS = bibprogs.o compiledate.o
 
 all:  libbibutils.a libbibprogs.a
 
-libbibutils.a: $(BIBUTILS_OBJS)
-	ar -cr libbibutils.a $(BIBUTILS_OBJS)
-	$(RANLIB) libbibutils.a
+libbibutils.a: libbibutils.la
+	ln -s .libs/libbibutils.a .
+
+libbibutils.la: $(BIBUTILS_LO)
+	libtool --mode=link  $(CC)  -version-info 0:0:0 -rpath /usr/lib -o libbibutils.la $(BIBUTILS_LO)
 
 libbibprogs.a: $(BIBPROGS_OBJS)
 	ar -cr libbibprogs.a $(BIBPROGS_OBJS)
@@ -39,6 +48,6 @@ clean:
 	/bin/rm -f *.o core
 
 realclean:
-	/bin/rm -f *.o *.a core
+	/bin/rm -rf *.o *.lo *.la *.a .libs core
 
 test:
-- 
tg: (b60e7a7..) debian/libtoolize-lib (depends on: debian/test-makefile)
