From: Pierre Gruet <pgtdebian@free.fr>
Date: Sat, 28 Mar 2020 13:34:04 +0100
Subject: Fixing spelling errors in manpage

---
 bibutils.dbk | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/bibutils.dbk b/bibutils.dbk
index 3726f42..e224033 100644
--- a/bibutils.dbk
+++ b/bibutils.dbk
@@ -690,12 +690,12 @@ enhancers",
     </refsect1>
     <refsect1>
       <title>License</title>
-      <para>All versions of bibutils are relased under the GNU Public
-	License (GPL). In a nutshell, feel free to download, run, and
-	modify these programs as required. If you re-release these, you
-	need to release the modified version of the source. (And I'd
-	appreciate patches as well...if you care enough to make the
-	change, then I'd like to see what you're adding or
+      <para>All versions of bibutils are released under the GNU General
+        Public License (GPL). In a nutshell, feel free to download, 
+        run, and modify these programs as required. If you re-release 
+        these, you need to release the modified version of the source. 
+        (And I'd appreciate patches as well...if you care enough to 
+        make the change, then I'd like to see what you're adding or
 	fixing.)</para>
       <para>
 	<ulink url="http://www.sourceforge.net/p/bibutils/home/Bibutils">
