From: Pierre Gruet <pgt@debian.org>
Date: Sun, 29 Mar 2020 10:47:03 +0200
Subject: Fixing a spelling error in lib/charsets.c

---
 lib/charsets.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/charsets.c b/lib/charsets.c
index 7a3c69f..a576d3e 100644
--- a/lib/charsets.c
+++ b/lib/charsets.c
@@ -34243,7 +34243,7 @@ allcharconvert_t allcharconvert[] = {
 	CHARSETARRAY( cp857 )
 },
 { "cp860",
-	"MS-DOS Code Page 860 Portugese",
+	"MS-DOS Code Page 860 Portuguese",
 	{ "CP860", "IBM860", "860", "", "", "", "", },
 	CHARSETARRAY( cp860 )
 },
